src/DynamoCoreWpf/Interfaces/IWatchHandler.cs

```diff
@@ -134,17 +134,17 @@ private WatchViewModel ProcessThing(SIUnit unit, ProtoCore.RuntimeCore runtimeCo
 
         private WatchViewModel ProcessThing(double value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
-            return new WatchViewModel(value.ToString(numberFormat, CultureInfo.InvariantCulture), tag, RequestSelectGeometry);
+            return new WatchViewModel(value, tag, RequestSelectGeometry);
         }
 
         private WatchViewModel ProcessThing(System.DateTime value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
-            return new WatchViewModel(value.ToString(PreferenceSettings.DefaultDateFormat, CultureInfo.InvariantCulture), tag, RequestSelectGeometry);
+            return new WatchViewModel(value, tag, RequestSelectGeometry);
         }
 
         private WatchViewModel ProcessThing(long value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
-            return new WatchViewModel(value.ToString(CultureInfo.InvariantCulture), tag, RequestSelectGeometry);
+            return new WatchViewModel(value, tag, RequestSelectGeometry);
```

Comment:
I see you moved some of these cases into the `StringFromObject` function if I am interpreting correctly, but what about `long`

Reply:
Long is also handled in [`GetStringFromObject` on `WatchViewModel`]( https://github.com/SHKnudsen/Dynamo/blob/a90c38c882eb9dc4f3a4140f1365148101647afa/src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs#L252)